src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -25,266 +25,234 @@
+  /** Generates a SamLocusIterator that will traverse over the relevant parts of the BAM. */
+  private def buildSamLocusIterator(in: SamReader, intervals: Option[IntervalList]): SamLocusIterator = {
+    val iterator = intervals match {
+      case None => new SamLocusIterator(in)
+      case Some(i) => new SamLocusIterator(in, i)
     }
 
-    // Get the reference bases
-    val referenceBases = referenceSequence.getBases
-
-    // We proceed through the read in the order it was aligned, but if the read is on the reverse strand, we will
-    // need to invert the read position since the last base in the read will be the first base sequenced
-    val readOffsetIncrement = if (rec.getReadNegativeStrandFlag) -1 else 1
-    val readLength = rec.getReadLength
-
-    // Get the read bases and qualities.
-    val read  = rec.getReadBases
-    val quals = rec.getBaseQualities
-
-    // For each alignment block
-    for (block <- rec.getAlignmentBlocks) {
-      var refPosition  = block.getReferenceStart - 1
-      var readPosition = block.getReadStart - 1
-      val readEnd      = readPosition + block.getLength
-      var readOffset   = if (rec.getReadNegativeStrandFlag) readLength - readPosition - 1 else readPosition
-
-      // Matches/mismatches
-      while (readPosition < readEnd) {
-        val readBase = read(readPosition)
-        val qual     = quals(readPosition)
-        val refBase  = referenceBases(refPosition)
-        incrementIfValidBase(readIndex=readIndex, readOffset=readOffset, readBase=readBase, qual=qual, refBase=refBase)
-        refPosition  += 1
-        readPosition += 1
-        readOffset   += readOffsetIncrement
-      }
-    }
+    val filters = new util.ArrayList[SamRecordFilter]()
+    filters.add(new SecondaryOrSupplementaryFilter)
+    filters.add(new FailsVendorReadQualityFilter)
+    if (!this.includeDuplicates) filters.add(new DuplicateReadFilter)
+    iterator.setSamFilters(filters)
+
+    iterator.setMappingQualityScoreCutoff(this.minMappingQuality)
+    iterator.setQualityScoreCutoff(this.minBaseQuality)
+    iterator.setEmitUncoveredLoci(false)
+    iterator.setIncludeIndels(false)
+    iterator
+  }
 
-    // Handle insertions
-    if (includeInsertions) {
-      var readOffset = if (rec.getReadNegativeStrandFlag) rec.getReadLength - 1 else 0
-      rec.getCigar.getCigarElements.foreach { elem =>
-        val op: CigarOperator = elem.getOperator
-        if (op == CigarOperator.INSERTION) {
-          var i = 0
-          while (i < elem.getLength) {
-            val qual = quals(readOffset)
-            increment(readIndex = readIndex, readOffset = readOffset, qual=qual)
-            readOffset += readOffsetIncrement
-            i += 1
-          }
+  /** Generates a variant mask object. */
+  private def buildVariantMask(variants: Option[PathToVcf], intervals: Option[IntervalList], dict: SAMSequenceDictionary): VariantMask = {
```

Comment:
I don't think VariantMask should deal with `Option[PathToVcf]` or `Option[IntervalList]`...